src/main/kotlin/org/jitsi/nlj/rtcp/RtcpRrGenerator.kt

```diff
@@ -104,15 +105,17 @@ class RtcpRrGenerator(
                     senderInfo.getDelaySinceLastSr(now)
                 ))
             }
+
+            val packets = mutableListOf<RtcpPacket>()
             if (reportBlocks.isNotEmpty()) {
-                // TODO send one compound packet.
-                RtcpRrPacketBuilder(reportBlocks = reportBlocks).build().apply {
-                    rtcpSender(this)
-                }
+                packets.add(RtcpRrPacketBuilder(reportBlocks = reportBlocks).build())
+            }
+            packets.addAll(additionalPacketSupplier())
 
-                additionalPacketSupplier().forEach {
-                    rtcpSender(it)
-                }
+            when (packets.size) {
+                0 -> {}
```

Comment:
You can just omit the `0` case altogether here.

Reply:
Wouldn't it fall back to the `else`?

Reply:
Oops, missed that.  You're right.